utilities/resolveComponents.js

```
@@ -0,0 +1,52 @@
+  if (!item?.components?.length) return item;
+
+  const map = toCatalogMap(catalog);
+  item.components = item.components.map((ref) => {
+    if (!ref?.uniqueName) return ref;
+    // Already resolved (has name / imageName / etc.)
+    if (ref.name || ref.imageName || ref.drops) {
+      return ref;
+    }
+    const entry = map.get(ref.uniqueName);
+    if (!entry) return ref;
+    const { parentUniqueNames: _parents, ...rest } = entry;
```

ESLint

[error] '_parents' is assigned a value but never used.

(no-unused-vars)